sys-kernel/gentoo-kernel-bin/gentoo-kernel-bin-6.4.4.ebuild

```diff
@@ -124,4 +135,14 @@ src_install() {
 		')' -delete || die
 	rm modprep/source || die
 	cp -p -R modprep/. "${ED}/usr/src/linux-${KPV}"/ || die
+
+	# We can only call linux-mod-r1_pkg_setup after we have
+	# setup our kernel sources
+	local KERNEL_DIR="${ED}/usr/src/linux-${KPV}"
+	linux-mod-r1_pkg_setup
+	modules_post_process
```

Comment:
Admit that I don't super like this, when making the eclass I specifically had the idea that this is not intended for the kernel itself (case in point the non-bin doesn't need it, but this ends up being a special case). It's not really setup as a utility eclass, and is supposed to be only for building modules.

I did think to make a non-phase function to replace pkg_setup at first but felt it didn't fit well with overriding linux-info's and ended up not bothering.

But oh well, guess could always rethink it later and I do see why you'd want to use it. So I'm not objecting.